cmake/build_helpers.cmake

```diff
@@ -17,4 +17,39 @@ function(ginkgo_compile_features name)
     if(GINKGO_WITH_IWYU AND GINKGO_IWYU_PATH)
         set_property(TARGET "${name}" PROPERTY CXX_INCLUDE_WHAT_YOU_USE ${GINKGO_IWYU_PATH})
     endif()
+    if(GINKGO_CHANGED_SHARED_LIBRARY)
+        # Put all shared libraries and corresponding imported libraries into the specficed path
+        set_property(TARGET "${name}" PROPERTY
+            RUNTIME_OUTPUT_DIRECTORY "${GINKGO_WINDOWS_SHARED_LIBRARY_PATH}")
+        set_property(TARGET "${name}" PROPERTY
+            ARCHIVE_OUTPUT_DIRECTORY "${GINKGO_WINDOWS_SHARED_LIBRARY_PATH}")
+        if(GINKGO_CHECK_PATH)
+            ginkgo_check_shared_library("${CMAKE_SHARED_LIBRARY_PREFIX}${name}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+        endif()
+    endif()
 endfunction()
+
+function(ginkgo_check_shared_library name)
+    set(PATH_LIST $ENV{PATH})
+    set(PASSED_TEST FALSE)
+    foreach(ITEM ${PATH_LIST})
```

Comment:
In modern CMake, this would rather be:
```cmake
foreach(ITEM IN LISTS PATH_LIST)
```

See: https://cmake.org/cmake/help/v3.0/command/foreach.html